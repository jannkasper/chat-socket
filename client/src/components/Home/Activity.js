import React from "react";
import Username from "../Username";
import Message from "../Message";
import Notice from "../Notice";

const Activity = ({ activity, scrollToBottom }) => {
    switch (activity.type) {
        case 'TEXT_MESSAGE':
            return <Message sender={activity.username} message={activity.text} timestamp={activity.timestamp}/>;
        case 'USER_ENTER':
            return (
                <Notice>
                    <div>
                        <Username key={0} username={activity.username}/> joined
                    </div>
                </Notice>
            );
        case 'USER_EXIT':
            return (
                <Notice>
                    <div>
                        <Username key={0} username={activity.username} /> left
                    </div>
                </Notice>
            );
        case 'TOGGLE_LOCK_ROOM':
            if (activity.locked) {
                return (
                    <Notice>
                        <div>
                            <Username key={0} username={activity.username} /> locked the room
                        </div>
                    </Notice>
                );
            } else {
                return (
                    <Notice>
                        <div>
                            <Username key={0} username={activity.username} /> unlocked the room
                        </div>
                    </Notice>
                );
            }
        default:
            return false;
    }
}

export default Activity