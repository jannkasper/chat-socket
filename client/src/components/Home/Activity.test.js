import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Activity from './Activity';
import { Provider } from 'react-redux';
import configureStore from '../../store';

const store = configureStore();

//jest.mock('components/T'); // Need store

describe('Activity component', () => {
  it('should display', () => {
    const activity = {
      type: '',
    };
    const { asFragment } = render(
      <Provider store={store}>
        <Activity activity={activity} scrollToBottom={jest.fn()} />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display TEXT_MESSAGE', () => {
    const activity = {
      type: 'TEXT_MESSAGE',
      username: 'alice',
      timestamp: new Date('2020-03-14T11:01:58.135Z').valueOf(),
      text: 'Hi!',
    };
    const { asFragment } = render(
      <Provider store={store}>
        <Activity activity={activity} scrollToBottom={jest.fn()} />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display USER_ENTER', () => {
    const activity = {
      type: 'USER_ENTER',
      username: 'alice',
    };
    const { asFragment } = render(
      <Provider store={store}>
        <Activity activity={activity} scrollToBottom={jest.fn()} />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display USER_EXIT', () => {
    const activity = {
      type: 'USER_EXIT',
      username: 'alice',
    };
    const { asFragment } = render(
      <Provider store={store}>
        <Activity activity={activity} scrollToBottom={jest.fn()} />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display TOGGLE_LOCK_ROOM', () => {
    const activity = {
      type: 'TOGGLE_LOCK_ROOM',
      locked: true,
      username: 'alice',
    };
    const { asFragment, rerender } = render(
      <Provider store={store}>
        <Activity activity={activity} scrollToBottom={jest.fn()} />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();

    activity.locked = false;

    rerender(
      <Provider store={store}>
        <Activity activity={activity} scrollToBottom={jest.fn()} />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
