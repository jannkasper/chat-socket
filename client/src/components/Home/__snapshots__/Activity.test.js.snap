// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity component should display 1`] = `<DocumentFragment />`;

exports[`Activity component should display TEXT_MESSAGE 1`] = `
<DocumentFragment>
  <div>
    <div
      class="chat-meta"
    >
      <span
        class="username"
        style="color: rgb(204, 241, 255);"
      >
        alice
      </span>
      <span
        class="muted timestamp"
      >
        12:01 PM
      </span>
    </div>
    <div
      class="chat"
    >
      Hi!
    </div>
  </div>
</DocumentFragment>
`;

exports[`Activity component should display TOGGLE_LOCK_ROOM 1`] = `
<DocumentFragment>
  <div>
    <div
      class="info"
    >
      <div>
        <span
          class="username"
          style="color: rgb(204, 241, 255);"
        >
          alice
        </span>
         locked the room
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Activity component should display TOGGLE_LOCK_ROOM 2`] = `
<DocumentFragment>
  <div>
    <div
      class="info"
    >
      <div>
        <span
          class="username"
          style="color: rgb(204, 241, 255);"
        >
          alice
        </span>
         unlocked the room
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Activity component should display USER_ENTER 1`] = `
<DocumentFragment>
  <div>
    <div
      class="info"
    >
      <div>
        <span
          class="username"
          style="color: rgb(204, 241, 255);"
        >
          alice
        </span>
         joined
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Activity component should display USER_EXIT 1`] = `
<DocumentFragment>
  <div>
    <div
      class="info"
    >
      <div>
        <span
          class="username"
          style="color: rgb(204, 241, 255);"
        >
          alice
        </span>
         left
      </div>
    </div>
  </div>
</DocumentFragment>
`;
